fedimint-server/src/db.rs

```diff
@@ -184,8 +184,6 @@ mod fedimint_migration_tests {
             .await;
 
         let _consensus_items = vec![ConsensusItem::Transaction(transaction)];
-
-        dbtx.commit_tx().await;
```

Comment:
This is now responsibility of a caller. It just makes more sense.

Reply:
Not committing felt wrong when I saw it because we drop the DBTX at the end of the scope and it's not obvious that it's just a pointer. I think in most other place we pass around mut refs anyway, so it isn't a problem. How do others think about this?